# Draft analysis 

---

Group name: Group C, Burcin Aksel & Nico Henzel

---


## Introduction

### Inhalte aus Project Proposal übernehmen und konsolidieren
### Burcin

*This section includes an introduction to the project motivation, data, and research question. Include a data dictionary* 

## Setup

### Anpassen, sobald Import benötigt wird
### Nico

In [1]:
import pandas as pd
import numpy as np
import altair as alt
import matplotlib as mlp
import xlrd # needed to read excel files. install xlrd module if import does not work!
import shutil # needed to copy files


## Data

## Nico

### Inhalte aus Data Correction und Project Proposal übernehmen

## Import data

In [2]:
# Declare variables
external_data = '..\\data\\external\\'
raw_data = '..\\data\\raw\\'

# File names
orig_income_file = 'Median_Household_Income_By_State_1990-2017.xls'
target_income_file = 'Median_Household_Income_By_State_1990-2017.csv'
orig_death_file = 'NCHS_-_Leading_Causes_of_Death__United_States.csv'


In [3]:
# Save external median income data as csv in folder 'raw'
# Read file
xls_household_file = pd.read_excel(external_data+orig_income_file)
# Save file
xls_household_file.to_csv(raw_data+target_income_file,index = None, header=True)
# Copy external leading cause of death data into folder 'raw'
shutil.copy(external_data+orig_death_file, raw_data+orig_death_file)


'..\\data\\raw\\NCHS_-_Leading_Causes_of_Death__United_States.csv'

In [17]:
# Declare both dataframes
df_income = pd.read_csv(raw_data+target_income_file)
df_death = pd.read_csv(raw_data+orig_death_file)

### Data structure

In [12]:
df_income.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68 entries, 0 to 67
Data columns (total 32 columns):
 #   Column                                                                              Non-Null Count  Dtype  
---  ------                                                                              --------------  -----  
 0   Table 102.30. Median household income, by state: Selected years, 1990 through 2017  59 non-null     object 
 1   Unnamed: 1                                                                          54 non-null     object 
 2   Unnamed: 2                                                                          54 non-null     object 
 3   Unnamed: 3                                                                          54 non-null     float64
 4   Unnamed: 4                                                                          52 non-null     float64
 5   Unnamed: 5                                                                          54 non-null     f

In [13]:
df_income

Table 102.30. Median household income, by state: Selected years, 1990 through 2017  \
0   [In constant 2017 dollars. Standard errors app...                                   
1                                               State                                   
2                                                   1                                   
3                           United States ...........                                   
4                        Alabama ....................                                   
..                                                ...                                   
63                         Wyoming ..................                                   
64  \1\Based on 1989 incomes collected in the 1990...                                   
65  \2\Based on 1999 incomes collected in the 2000...                                   
66  NOTE: Constant dollars adjusted by the Consume...                                   
67  SOURCE: U.S. Department of Commerce, Census Bu...                                   

   Unnamed: 1 Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  Unnamed: 6  \
0         NaN        NaN         NaN         NaN         NaN         NaN   
1     1990\1\    2000\2\      2005.0         NaN      2010.0         NaN   
2           2          3         4.0         NaN         5.0         NaN   
3       57500      62000     58200.0        80.0     56400.0        40.0   
4       45200      50400     46400.0       400.0     45600.0       320.0   
..        ...        ...         ...         ...         ...         ...   
63      51900      55900     58100.0      1160.0     60300.0      1300.0   
64        NaN        NaN         NaN         NaN         NaN         NaN   
65        NaN        NaN         NaN         NaN         NaN         NaN   
66        NaN        NaN         NaN         NaN         NaN         NaN   
67        NaN        NaN         NaN         NaN         NaN         NaN   

    Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 22  Unnamed: 23  \
0          NaN         NaN         NaN  ...          NaN          NaN   
1       2013.0         NaN      2014.0  ...          NaN          NaN   
2          6.0         NaN         7.0  ...          NaN          NaN   
3      55100.0        40.0     55600.0  ...          NaN          NaN   
4      45200.0       410.0     44400.0  ...          NaN          NaN   
..         ...         ...         ...  ...          ...          ...   
63     61900.0      1150.0     59100.0  ...          NaN          NaN   
64         NaN         NaN         NaN  ...          NaN          NaN   
65         NaN         NaN         NaN  ...          NaN          NaN   
66         NaN         NaN         NaN  ...          NaN          NaN   
67         NaN         NaN         NaN  ...          NaN          NaN   

    Unnamed: 24  Unnamed: 25  Unnamed: 26  Unnamed: 27  Unnamed: 28  \
0           NaN          NaN          NaN          NaN          NaN   
1           NaN          NaN          NaN          NaN          NaN   
2           NaN          NaN          NaN          NaN          NaN   
3           NaN          NaN          NaN          NaN          NaN   
4           NaN          NaN          NaN          NaN          NaN   
..          ...          ...          ...          ...          ...   
63          NaN          NaN          NaN          NaN          NaN   
64          NaN          NaN          NaN          NaN          NaN   
65          NaN          NaN          NaN          NaN          NaN   
66          NaN          NaN          NaN          NaN          NaN   
67          NaN          NaN          NaN          NaN          NaN   

    Unnamed: 29  Unnamed: 30  Unnamed: 31  
0           NaN          NaN          NaN  
1           NaN          NaN          NaN  
2           NaN          NaN               
3           NaN          NaN          NaN  
4           NaN          NaN          NaN  
..          ...          ..

In [14]:
df_death.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10868 entries, 0 to 10867
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     10868 non-null  int64  
 1   113 Cause Name           10868 non-null  object 
 2   Cause Name               10868 non-null  object 
 3   State                    10868 non-null  object 
 4   Deaths                   10868 non-null  int64  
 5   Age-adjusted Death Rate  10868 non-null  float64
dtypes: float64(1), int64(2), object(3)
memory usage: 509.6+ KB


In [15]:
df_death

Year                                     113 Cause Name  \
0      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
1      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
2      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
3      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
4      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
...     ...                                                ...   
10863  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10864  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10865  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10866  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10867  1999  Nephritis, nephrotic syndrome and nephrosis (N...   

                   Cause Name          State  Deaths  Age-adjusted Death Rate  
0      Unintentional injuries  United States  169936                     49.4  
1      Unintentional injuries        Alabama    2703                     53.8  
2      Unintentional injuries         Alaska     436                     63.7  
3      Unintentional injuries        Arizona    4184                     56.2  
4      Unintentional injuries       Arkansas    1625                     51.8  
...                       ...            ...     ...                      ...  
10863          Kidney disease       Virginia    1035                     16.9  
10864          Kidney disease     Washington     278                      5.2  
10865          Kidney disease  West Virginia     345                     16.4  
10866          Kidney disease      Wisconsin     677                     11.9  
10867          Kidney disease        Wyoming      30                      6.8  

[10868 rows x 6 columns]

In the death dataset we have 10868 cases and 6 columns.

The columns are:
* years, from 1999 to 2017
* 113 cause name, the NDI ICD-10 113 categories for causes of death
* cause name, the generic name for the death cause defined in the 113 cause name column
* state, in which state the data was collected
* death, the count of the total deaths 
* age-adjusted death rate, the standardized death percentage for the specific state in the observed year.


The following steps are:

* The income dataset needs to be cleaned and corrected in order to be joined with the death dataset (see data-corrections)
* The columns for the income dataset need to be transformed in order to have the same format as the death dataset.
* The join can be done by state, the missing income values for the years not represented in the income dataset need to either be dropped or replaced


### Data corrections

In [ ]:
# Data preparation

We prepared the income data set in order to use it for our project. 
The original .xls data can be found in the data folder under external or the .csv under raw 

In the final death data set we have 468 cases and 3 columns.

The columns are:
* state, in which the data was collected
* years, 1990, 2000, 2005, 2010 and 2013-2017
* median_household_income

In [16]:
# JOIN

After joining, the median_household_income will be an additional column for the death data set.

This will leave us with 6 columns in total, which we can use for our data analysis (we will disregard the 113 Cause name column since the same information is provided by the cause name column)

### Variable lists
## Burcin

In [ ]:
# auf Fertigem Datensatz Variablen auflisten

### Data splitting

## Burcin

## Analysis

## Nico

Analysis Approach umsetzen

### Descriptive statistics

### Exploratory data analysis

### Relationships

## Model

## Nico
### Kopieren aus Notebook 44 und Modeltypen passend zu proposal umsetzen

Abstimmung, welche Metriken und Erkenntnisse aus den Modellen für uns relevant sind

### Select model

### Training and validation

### Fit model

### Evaluation on test set

### Save model



Save your model in the folder `models/`. Use a meaningful name and a timestamp.

## Conclusions

Am Ende die weiteren Schritte und bisherigen Interprationen auflisten.